library/train_util.py

```diff
@@ -5202,7 +5203,18 @@ def sample_images_common(
     if distributed_state.num_processes <= 1:
         # If only one device is available, just use the original prompt list. We don't need to care about the distribution of prompts.
         with torch.no_grad():
+            idx = 0
             for prompt_dict in prompts:
+                if '__caption__' in prompt_dict.get("prompt") and example_tuple:
+                    while example_tuple[1][idx] == '':
+                        idx = (idx + 1) % len(example_tuple[1])
+                        if idx == 0:
+                            break
+                    prompt_dict["prompt"] = prompt_dict.get("prompt").replace('__caption__', 'example_tuple[1][idx]') 
```

Comment:
Should this be:
 `prompt_dict["prompt"] = prompt_dict.get("prompt").replace('__caption__', example_tuple[1][idx])`?

Current behavior replacing with the string 'example_tuple[1][idx]' instead of the value of the tuple